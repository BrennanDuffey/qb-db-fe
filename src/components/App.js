import React from 'react';
import logo from './logo.svg';
import { Route } from 'react-router-dom';

function App() {
  return (
    <Main className="">
      <Route exact path='/' component={SplashPage}/>
    </Main>
  );
}

export default App;
