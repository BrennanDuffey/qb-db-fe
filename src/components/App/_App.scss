
.app {
  text-align: center;
  margin: 0;
  padding: 0;
  height: 100vh;
  header {
    background-color: rgba(0, 0, 0, .1);
    border-bottom: solid 1px rgba(0, 0, 0, .15);
    display: flex;
  }
  .nav {
    text-decoration: none;
    color: black;
    padding: 10px 15px 5px;
    border-bottom: solid 2px rgba(0, 0, 0, .1);
    &:hover {
      border-bottom: solid 2px rgba(0, 0, 0, .4);
    }
  }
  section {
    padding: 20px 0;
    width: 80%;
    margin: 0 auto;
    h2 {
      background-color: rgba(0, 0, 0, .1);
      padding: 10px;
      border: solid 1px rgba(0, 0, 0, .15);
      border-radius: 12px;
      width: 60%;
      margin: auto;
      &:hover {
        border: solid 1px rgba(0, 0, 0, .25);
        background-color: rgba(0, 0, 0, .2);
      }
    }
  }
}

main {
  text-align: center;
}