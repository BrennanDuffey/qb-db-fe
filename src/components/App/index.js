import React from 'react';
import { Route } from 'react-router-dom';
import SplashPage from '../SplashPage';
import Controls from '../../containers/Controls';
import QuizContainer from '../../containers/QuizContainer';
import StudyContainer from '../../containers/StudyContainer';



function App() {
  return (
    <main className="">
      <button>Stuff </button>
      <Route exact path='/' component={SplashPage} />
      <Route exact path='/Controls' component={Controls} />
      <Route exact path='/Quiz' component={QuizContainer} />
      <Route exact path='/Study' component={StudyContainer} />
    </main>
  );
}

export default App;
