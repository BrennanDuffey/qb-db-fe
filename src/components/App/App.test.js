import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';
import { shallow, mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';

describe('App', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <App />
      </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('shallow render should match snapshot', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot when fully mounted', () => {
    const wrapper = mount(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
