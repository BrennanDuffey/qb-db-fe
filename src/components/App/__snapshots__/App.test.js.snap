// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App shallow render should match snapshot 1`] = `ShallowWrapper {}`;

exports[`App should match snapshot when fully mounted 1`] = `ReactWrapper {}`;
